File: Facade/PHP/VegNonBothRestaurant.php
<?php

require_once "IRestaurant.php";
require_once "Menu.php";

class VegNonBothRestaurant implements IRestaurant {
    private Menu $menu;

    public function __construct()
    {
        $this->menu = new Menu("Veg Non Both Menu");
    }

    public function getMenus() {
        return $this->menu;
    }
}

File: Facade/Dart/VegNonBothRestaurant.dart
import 'IRestaurant.dart';
import 'Menu.dart';

class VegNonBothRestaurant implements IRestaurant {
  Menu? _menu;

  VegNonBothRestaurant() {
    _menu = Menu("Veg Non Both Menu");
  }

  @override
  Menu? getMenus() {
    return _menu;
  }
}


